clippy_lints/src/escape.rs

```diff
@@ -205,4 +209,13 @@ impl<'a, 'tcx> EscapeDelegate<'a, 'tcx> {
             false
         }
     }
+
+    fn is_unsized_box(&self, ty: Ty<'tcx>, span: Span) -> bool {
+        if !ty.is_box() {
+            return false;
+        }
+
+        !ty.boxed_ty()
```

Comment:
IMO this function is clearer if you just write `ty.is_box() && !ty.boxed_ty().is_sized(..)`

Everything else LGTM.